nvflare/lighter/entity.py

```diff
@@ -284,27 +315,50 @@ def get_listening_host(self) -> Optional[ListeningHost]:
         return lh
 
     def get_connect_to(self) -> Optional[ConnectTo]:
+        """Get the connect_to property of the participant
+
+        Returns: a ConnectTo object
+
+        """
         h = self.get_prop(PropKey.CONNECT_TO)
         if not h:
             return None
         else:
             return parse_connect_to(h)
 
 
-def participant_from_dict(participant_def: dict, project=None) -> Participant:
-    name = participant_def.pop("name", None)
-    if not name:
-        raise ValueError("missing participant name")
+def _must_get(d: dict, key: str):
+    """Must get property of the specified key from the dict
 
-    type = participant_def.pop("type", None)
-    if not type:
-        raise ValueError("missing participant type")
+    Args:
+        d: the dict that contains participant properties
+        key: key of the property to get
+
+    Returns: the value of the property. If the property does not exist, ValueError exception is raised.
 
-    org = participant_def.pop("org", None)
-    if org:
-        raise ValueError("missing participant org")
+    """
+    v = d.pop(key, None)
+    if not v:
+        raise ValueError(f"missing participant {key}")
+    return v
```

Comment:
this ```_must_get``` method is equivalent as just:

v = d.pop(key)

just that using pop will raise "KeyError"